### PR TITLE
Instantiate record values using closures values captured is typedesc

### LINKED ISSUE #22908: Capture closure variables in typedesc
Sub task of #20851

Change folloing
```
    %2 = NewMap map<any | error{string, $anonType$builtin$0}>;
    %6 = ConstLoad 20;
    %5 = <any | error> %6;
    %8 = ConstLoad x;
    %2[%8] = %5;
    %7 = newType Foo;
    %8 = NewMap %7;
```
to
```
    %2 = NewMap map<any | error{string, $anonType$builtin$0}>;
    %6 = ConstLoad 20;
    %5 = <any | error> %6;
    %8 = ConstLoad x;
    %2[%8] = %5;
    %7 = newType Foo(%2);
    %8 = NewMap %7;
```

### PR DESCRIPTION
**Pull request by manuranga**
Fixes #22908 sub task of #20851

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java [hunk 1]
```diff
@@ -568,17 +573,80 @@ private void createInstantiateMethod(ClassWriter cw, BRecordType recordType) {
...
         mv.visitEnd();
     }
 
+    private StringBuilder calcClosureMapSignature(int size) {
+        StringBuilder closureParamSignature = new StringBuilder();
+        for (int i = 0; i < size; i++) {
+            closureParamSignature.append('L');
+            closureParamSignature.append(MAP_VALUE);
+            closureParamSignature.append(";Z");
+
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java:645]: remove this new line

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java [hunk 2]
```diff
@@ -1245,6 +1249,13 @@ public void visit(BLangRecordLiteral.BLangMapLiteral mapLiteral) {
 
     @Override
     public void visit(BLangRecordLiteral.BLangStructLiteral structLiteral) {
+        SymbolEnv symbolEnv = env.createClone();
+//        bLangLambdaFunction.capturedClosureEnv = symbolEnv;
+        BLangFunction enclInvokable = (BLangFunction) symbolEnv.enclInvokable;
+        // Save param closure map of the encl invokable.
+//        bLangLambdaFunction.paramMapSymbolsOfEnclInvokable = enclInvokable.paramClosureMap;
+//        boolean isWorker = bLangLambdaFunction.function.flagSet.contains(Flag.WORKER);
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java:1257]: commented out sections? do we need them?

Comment (manuranga) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java:1257]: no, my bad, I though I removed before committing. will remove.

**Comment by Kishanthan** [comment]
LGTM

**Pull request merged**